src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVelesTracking.java

```diff
@@ -0,0 +1,138 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.annotations.NetworkTrafficDump;
+import com.wikia.webdriver.common.core.url.Page;
+import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import net.lightbody.bmp.core.har.HarEntry;
+import net.lightbody.bmp.filters.RequestFilter;
+import net.lightbody.bmp.util.HttpMessageContents;
+import net.lightbody.bmp.util.HttpMessageInfo;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.testng.annotations.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+
+public class TestAdsVelesTracking extends NewTestTemplate {
+
+  public static final String VELES_TRACKING_REQUEST_PART = "adengadinfo";
+
+  public static final String VELES_TRACKING_POSITION_PARAMETER = "kv_pos";
+
+  public static final String VELES_TRACKING_PRICE_PARAMETER = "bidder_8";
+
+  @NetworkTrafficDump
+  @Test(
+      groups = "AdsTrackingVeles",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVelesTracking")
+  public void adsTrackingVelesTracked(final Page page, final Map<String, String> positionsAndPrices) {
+    networkTrafficInterceptor.addRequestFilter(new RequestFilter() {
+      @Override
+      public HttpResponse filterRequest(HttpRequest request, HttpMessageContents contents,
+                                        HttpMessageInfo messageInfo) {
+
+        if(request.getUri().contains("ads")){
+          DefaultFullHttpResponse
+              response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.INTERNAL_SERVER_ERROR);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1488)